x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/ingest/InferenceProcessor.java

```diff
@@ -128,8 +129,8 @@ void mutateDocument(InferModelAction.Response response, IngestDocument ingestDoc
             throw new ElasticsearchStatusException("Unexpected empty inference response", RestStatus.INTERNAL_SERVER_ERROR);
         }
         response.getInferenceResults().get(0).writeResult(ingestDocument, this.targetField);
-        if (modelInfoField != null) {
-            ingestDocument.setFieldValue(modelInfoField, modelInfo);
+        if (includeModelMetadata) {
+            ingestDocument.setFieldValue(modelInfoField + "." + MODEL_ID, modelId);
```

Comment:
It seems that `modelInfoField` should never be `null`?
Does it make sense to add `requireNonNull` in the constructor?